Examples from BehaviorTree.CPP tutorial

stepkun opened the pull request:
Implementation draft of the tutorial examples from [behaviortree.dev.](https://www.behaviortree.dev/).

Not all of them are working yet. See README.md in examples folder.

Enhanced Cargo.toml to exclude the examples on `crates.io` and usage of crate `rand` for the examples.

stepkun committed:
first draft of examples

stepkun committed:
fixed comments

ncrothers commented:
Hi, I'm sorry about leaving your PRs open, I do appreciate you implementing the examples! The reason I hadn't merged it or left comments is because I was busy with trying to finish up a "proper" release of the library in a state that I was happy with, which completely changed the API for the project. It was constantly changing while I was working on it so I didn't want to attempt to merge your code in until I was finished with it. Once I have that stabilized, I would love to incorporate your changes in!

stepkun commented:
Hi, I dropped the PR and removed the repo, as I realized that your implementation moved on and some effort is necessary to adopt the PR to your ongoing development.
I started an independent [behaviortree](https://github.com/stepkun/behaviortree) implementation. There the examples are available.
